XamarinCommunityToolkit/Behaviors/MaxLengthReachedBehavior.shared.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.ComponentModel;
+using System.Windows.Input;
+using Xamarin.CommunityToolkit.Helpers;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Behaviors
+{
+	public class MaxLengthReachedBehavior : BaseBehavior<InputView>
+	{
+		public static readonly BindableProperty CommandProperty
+			= BindableProperty.Create(nameof(Command), typeof(ICommand), typeof(MaxLengthReachedBehavior));
+
+		public ICommand Command
+		{
+			get => (ICommand)GetValue(CommandProperty);
+			set => SetValue(CommandProperty, value);
+		}
+
+		public static readonly BindableProperty ShouldDismissKeyboardAutomaticallyProperty
+			= BindableProperty.Create(nameof(ShouldDismissKeyboardAutomatically), typeof(bool), typeof(MaxLengthReachedBehavior), false);
+
+		public bool ShouldDismissKeyboardAutomatically
+		{
+			get => (bool)GetValue(ShouldDismissKeyboardAutomaticallyProperty);
+			set => SetValue(ShouldDismissKeyboardAutomaticallyProperty, value);
+		}
+
+		readonly WeakEventManager maxLengthReachedEventManager = new WeakEventManager();
+
+		public event EventHandler<MaxLengthReachedEventArgs> MaxLengthReached
+		{
+			add => maxLengthReachedEventManager.AddEventHandler(value);
+			remove => maxLengthReachedEventManager.RemoveEventHandler(value);
+		}
+
+		protected override void OnViewPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			base.OnViewPropertyChanged(sender, e);
+			if (e.PropertyName == InputView.TextProperty.PropertyName)
+				OnTextPropertyChanged();
+		}
+
+		void OnTextPropertyChanged()
+		{
+			if (!(View.Text?.Length >= View.MaxLength))
```

Comment:
Should we verify if the value is less than zero and if it's then throw an Exception?